File: Java Based Archives/TwoWayDataBinding/app/src/main/java/com/android/twowaydatabinding/MainActivity.java
package com.android.twowaydatabinding;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;

import android.os.Bundle;

import com.android.twowaydatabinding.databinding.ActivityMainBinding;
import com.android.twowaydatabinding.entity.Contact;

public class MainActivity extends AppCompatActivity {
    private ActivityMainBinding activityMainBinding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Contact contact = new Contact("Alex", "alex@gmail.com");

        activityMainBinding = DataBindingUtil.setContentView(this, R.layout.activity_main);
        activityMainBinding.setContact(contact);

    }
}

File: Views/LiveDataFundamentals2.0/app/src/main/java/com/example/viewmodeldemo2/MainActivity.kt
package com.example.viewmodeldemo2

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.databinding.DataBindingUtil
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import com.example.viewmodeldemo2.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private lateinit var viewModel: MainActivityViewModel
    private lateinit var viewModelFactory: MainActivityViewModelFactory
    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        // Initialize DataBinding
        binding = DataBindingUtil.setContentView(this, R.layout.activity_main)
        // Initialize ViewModelFactory with startingTotal value
        viewModelFactory = MainActivityViewModelFactory(0)
        // Get an instance of ViewModel using ViewModelProvider with ViewModelFactory
        viewModel = ViewModelProvider(this, viewModelFactory)[MainActivityViewModel::class.java]

        binding.myViewModel = viewModel

        // Observe changes to totalData and update the UI
        viewModel.totalData.observe(this, Observer { it ->
            binding.resultTextView.text = "Total: $it"
        })
    }
}
